src/mon/Monitor.cc

```diff
@@ -1433,12 +1433,14 @@ void Monitor::handle_sync_get_chunk(MonOpRequestRef op)
   while (sp.last_committed < paxos->get_version() && left > 0) {
     bufferlist bl;
     sp.last_committed++;
-    store->get(paxos->get_name(), sp.last_committed, bl);
-    // TODO: what if store->get returns error or empty bl?
-    tx->put(paxos->get_name(), sp.last_committed, bl);
-    left -= bl.length();
-    dout(20) << __func__ << " including paxos state " << sp.last_committed
+
+    int err = store->get(paxos->get_name(), sp.last_committed, bl);
+    if (err == 0 && bl.length() > 0) {
+      tx->put(paxos->get_name(), sp.last_committed, bl);
+      left -= bl.length();
+      dout(20) << __func__ << " including paxos state " << sp.last_committed
 	     << dendl;
+    }
```

Comment:
I'm afraid this does not fix the issue. The question we should be asking is "what should we do if `store->get()` returns error or an empty bufferlist?", instead of only handling it if we have a successful call.